be/src/env/env_s3.cpp

```diff
@@ -77,11 +77,16 @@ StatusOr<std::unique_ptr<RandomAccessFile>> EnvS3::new_random_access_file(const
     std::string object = path.substr(namenode.size(), path.size() - namenode.size());
     Aws::Client::ClientConfiguration config;
     config.scheme = Aws::Http::Scheme::HTTP;
-    if (!config::aws_s3_endpoint.empty()) {
+    assert(!config::aws_s3_endpoint.empty());
+    if (is_oss_path(namenode.c_str())) {
+        config.endpointOverride = get_endpoint_from_oss_bucket(bucket, config::aws_s3_endpoint);
+    } else {
         config.endpointOverride = config::aws_s3_endpoint;
     }
     config.maxConnections = config::aws_s3_max_connection;
     S3Credential cred;
+    assert(!config::aws_access_key_id.empty());
```

Comment:
for aws environment,  `aws_access_key_id` could be empty. If its empty, it will use default credential file, usually it's ~/.aws/credentials.

Reply:
done